lib/vagrant-google/action/run_instance.rb

```diff
@@ -287,19 +288,35 @@ def call(env) # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
             env[:interrupted] = true
           end
 
+          # Parse out the image project in case it was not set
+          # and check if it is part of a public windows project
+          img_project = image.split("/")[6]
+          is_windows_image = img_project.eql?("windows-cloud") || img_project.eql?("windows-sql-cloud")
+
+          # Reset the password if a windows image unless flag overrides
+          setup_winrm_password = zone_config.setup_winrm_password
+          if setup_winrm_password.nil? && is_windows_image
+            setup_winrm_password = true
+          end
+
+          if setup_winrm_password
+            env[:ui].info("Setting up WinRM Password")
+            env[:action_runner].run(Action.action_setup_winrm_password, env)
+          end
+
           unless env[:terminated]
```

Comment:
So while debugging I noticed that winrm options are not really set:
```
ls @machine.winrm_info
NoMethodError: undefined method `winrm_info' for #<Vagrant::Machine: winserver (VagrantPlugins::Google::Provider)>
from /Users/temikus/.rbenv/versions/2.4.3/lib/ruby/gems/2.4.0/gems/pry-0.12.2/lib/pry/core_extensions.rb:17:in `__pry__'
```

I suspect it may be because we never set up the winrm capability in the provider, like Azure did:
```
https://github.com/Azure/vagrant-azure/blob/97f4d125aebcb5669e9d1278b702e13554bef640/lib/vagrant-azure/capabilities/winrm.rb
```

Can you take a look?

Reply:
I had originally had all the features from the Azure plugin included, like this capability file but refactored it out when it didn't seem to do anything.  I am confused as to what the difference is between including the details in the config (e.g. config.winrm.*) and the need for setting up `env[:machine_winrm_info]` via this capability.  

The WinRM communicator does not appear to need this winrm_info, using the details from the Vagrantfile config instead.  However, I do see that there is a `machine_ssh_info`, so it would make sense to mimic that with WinRM.

Reply:
After some more digging, I've found that if I add in a `read_winrm_info` functionality, I am actually breaking things.  I've found that in the following file, it has a default way of computing winrm_info if it is not explicitly defined.  That must be why it works without adding in this functionality.

https://github.com/hashicorp/vagrant/blob/2c4a40fccbcd1869c7ea457d6f251c67c30eda71/plugins/communicators/winrm/helper.rb#L9

Would this be sufficient? Else I can explicitly add in this functionality, though I am still troubleshooting it.

Reply:
I should've been more clear here, sorry m(_ _)m

So, ideally we would want the provider to explicitly set that guest capability to make sure we don't rely on heuristics. However, if it's too difficult I'm also open leaving that as an explicit TODO in the code and creating an issue for the future.

WDYT?

Reply:
I'll take another shot at implementing this and see if I can get it working correctly.

Reply:
Let's go ahead and use the default heuristics.